File: 2018/11/abc/abc.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const lines = [];

rl.on("line", line => {
  lines.push(line);
});

rl.on("close", () => {
  const nums = lines[0]
    .split(" ")
    .map(n => parseInt(n))
    .sort((a, b) => a - b);

  const chars = lines[1].split("");
  const lookup = {};
  const results = [];

  for (const i in nums) {
    const num = nums[i];
    const letter = String.fromCharCode(65 + i * 1);
    lookup[letter] = num;
  }

  for (const char of chars) {
    results.push(lookup[char]);
  }

  console.log(results.join(" "));
});

File: 2018/11/abc/abc.py
import sys

lines = sys.stdin.readlines()
nums_orig = list(map(int, lines[0].split(' ')))
nums_sort = sorted(nums_orig)
chars = list(lines[1].strip())
mapping = {}
for i, n in enumerate(nums_sort):
    mapping[chr(65 + i)] = n
output = ' '.join([str(mapping[char]) for char in chars])
print(output)
